sycl/include/sycl/ext/oneapi/experimental/annotated_usm/alloc_base.hpp

```diff
@@ -0,0 +1,264 @@
+//==-------- alloc_base.hpp - SYCL annotated usm basic allocation -------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <sycl/ext/oneapi/experimental/annotated_ptr/annotated_ptr.hpp>
+#include <sycl/ext/oneapi/experimental/annotated_usm/alloc_util.hpp>
+
+#define VALIDATE_PROPERTIES(t)                                                 \
```

Comment:
Macros are generally not recommended to be used with modern C++. Is this design really necessary?

Reply:
Macro removed